### PR TITLE
[plugin-web-app] Add ability to capture http response in proxy

### LINKED ISSUE #1772: Ability to save response body to the variable to check it
**Type:** Feature request

**Goal:** Add ability to manage response body when the system uses Proxy functionality and finds the particular request

**Description:** Now we can manage only request content and deal with it using several commands such as 
`When I capture HTTP GET request with URL pattern text and save request data to SCENARIO variable request data`

In the future, we would like to have almost the same commands, but for **response** data. It can help to work with many requests and check it using Vividus.

Just small example:
image

### PR DESCRIPTION
**Pull request by uarlouski**
Closes #1772

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-web-app.adoc [hunk 1]
```diff
@@ -698,18 +697,30 @@ The actions preformed by the step:
...
 ----
-When I capture HTTP $httpMethods request with URL pattern `$urlPattern` and save URL to $scopes variable `$variableName`
+When I capture HTTP $httpMethods request with URL pattern `$urlPattern` and save $httpMessagePart to $scopes variable `$variableName`
 ----
 * `$httpMethods` - The "or"-separated set of HTTP methods to filter by, e.g. `GET or POST or PUT`.
 * `$urlPattern` - The regular expression to match HTTP request URL.
+* `$httpMessagePart` - the HTTP message part to save. One of:
+** `URL` - the request URL
+** `URL query` - the request URL query parameters
+** `request data` - the request data includes the following keys:
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-web-app.adoc:713]:
```suggestion
* `$httpMessagePart` - the HTTP message part to save. One of:
** `URL` - The request URL.
** `URL query` - The request URL query parameters.
** `request data` - The request data includes the following keys:
```

File: docs/modules/plugins/pages/plugin-web-app.adoc [hunk 2]
```diff
@@ -698,18 +697,30 @@ The actions preformed by the step:
...
+*** `query` - The URL query is stored as a collection of key and value pairs, where key is the name of the query parameter and value is the list of query parameter values. The query parameter values are accessible via zero-based index.
+*** `requestBody.mimeType` - The MIME type of posted data, the variable will not be saved if MIME type is not present.
+*** `requestBody.text` - The posted data as plain text, the variable will not be saved if the request body is not present.
+*** `requestBodyParameters` - The form data parameters are stored as a collection of key and value pairs, where key is the name of the form parameter and value is the list of form parameter values. The form parameter values are accessible via zero-based index.
+*** `responseStatus` - The response status, the variable will not be saved if the response is not present.
+** `response data` - the response data includes the following keys:
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-web-app.adoc:719]:
```suggestion
** `response data` - The response data includes the following keys:
```

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/model/HttpMessagePart.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    public abstract Object get(HarEntry harEntry);
+
+    private static Map<String, List<String>> getQueryParameters(HarRequest request)
+    {
+        return request.getQueryString().stream().collect(
+                groupingBy(HarQueryParam::getName, mapping(HarQueryParam::getValue, toList())));
+    }
+
+    private static Map<String, Object> extractFullRequestData(HarEntry harEntry)
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/model/HttpMessagePart.java:78]: it can be inlined

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-web-app.adoc [hunk 4]
```diff
@@ -698,18 +697,30 @@ The actions preformed by the step:
...
 
 [source,gherkin]
 ----
-When I capture HTTP $httpMethods request with URL pattern `$urlPattern` and save URL to $scopes variable `$variableName`
+When I capture HTTP $httpMethods request with URL pattern `$urlPattern` and save $httpMessagePart to $scopes variable `$variableName`
 ----
 * `$httpMethods` - The "or"-separated set of HTTP methods to filter by, e.g. `GET or POST or PUT`.
 * `$urlPattern` - The regular expression to match HTTP request URL.
+* `$httpMessagePart` - The HTTP message part to save. One of:
+** `URL` - The request URL
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-web-app.adoc:711]:
```suggestion
** `URL` - The request URL.
```

File: docs/modules/plugins/pages/plugin-web-app.adoc [hunk 5]
```diff
@@ -698,18 +697,30 @@ The actions preformed by the step:
...
+** `URL query` - The request URL query parameters
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-web-app.adoc:712]:
```suggestion
** `URL query` - The request URL query parameters.
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-web-app.adoc (hunk 4 above)
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-web-app.adoc:711]: Friday evening :)

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**